src/tsdb/src/tsdbMain.c

```diff
@@ -106,6 +107,17 @@ STsdbRepo *tsdbOpenRepo(STsdbCfg *pCfg, STsdbAppH *pAppH) {
     return NULL;
   }
 
+  if(pCfg->update == TD_ROW_PARTIAL_UPDATE) {
+    SMergeBuf buf;
+    if((buf = tsdbMakeBuf(512)) == NULL) {
+      tsdbError("vgId:%d failed to open TSDB repository while opening merge buffer pool since %s", config.tsdbId,
```

Comment:
We can avoid allocating memory here by using taosTMalloc series APIs. Allocate memory when it is really neccessary.